packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,224 @@
+
+  /**
+   * The configuration of the provisioning artifact (also known as a version).
+   */
+  readonly provisioningArtifacts: ProvisioningArtifactProperties[];
+
+  /**
+   * ReplaceProvisioningArtifacts
+   * @default false
+   */
+  readonly replaceProvisioningParameters?: boolean;
+
+  /**
+   * The support information about the product
+   * @default - No support description provided
+   */
+  readonly supportDescription?: string;
+
+  /**
+   * The contact email for product support.
+   * @default - No support email provided
+   */
+  readonly supportEmail?: string;
+
+  /**
+   * The contact URL for product support.
+   * @default - No support URL provided
+   */
+  readonly supportUrl?: string;
+}
+
+/**
+ * A Service Catalog Cloudformation Product.
+ * @resource AWS::ServiceCatalog::CloudFormationProduct
+ */
+export class Product extends ProductBase {
+  /**
+   * Creates a Product construct that represents an external product.
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param attrs Product import properties
+   */
+  public static fromProductAttributes(scope: Construct, id: string, attrs: ProductAttributes): IProduct {
+    const parts = Stack.of(scope).parseArn(attrs.productArn);
+    function throwError(errorMessage: string): never {
+      throw new Error(errorMessage);
+    }
+    class Import extends ProductBase {
+      public readonly productId = parts.resourceName ?? throwError('Product arn missing Product ID during import from attributes');
+      public readonly productArn = attrs.productArn;
+      public readonly productName = attrs.productName;
+    }
+    return new Import(scope, id);
+  }
+
+  public readonly productArn: string;
+  public readonly productName: string;
+  public readonly productId: string;
+  private readonly product: CfnCloudFormationProduct;
+
+  constructor(scope: Construct, id: string, props: ProductProps) {
+    super(scope, id, {});
+
+    this.validateProductProps(props);
+
+    this.product = new CfnCloudFormationProduct(this, 'Resource', {
+      acceptLanguage: props.acceptLanguage,
+      description: props.description,
+      distributor: props.distributor,
+      name: props.productName,
+      owner: props.owner,
+      provisioningArtifactParameters: this.generateProvisioningArtifactParameters(props),
+      replaceProvisioningArtifacts: props.replaceProvisioningParameters,
+      supportDescription: props.supportDescription,
+      supportEmail: props.supportEmail,
+      supportUrl: props.supportUrl,
+    });
+
+    this.productArn = this.getResourceArnAttribute(this.product.ref, {
+      service: 'catalog',
+      resource: 'product',
+      resourceName: this.product.ref,
+    });
+
+    this.productName = this.getResourceNameAttribute(props.productName);
+    this.productId = this.getResourceNameAttribute(this.product.ref);
+  }
+
+  private generateProvisioningArtifactParameters(props: ProductProps): any[] {
```

Comment:
Why does this return `any[]`? It should return `CfnCloudFormationProduct.ProvisioningArtifactPropertiesProperty[]`, right?